File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-4.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> res = new ArrayList<>();
        if(root == null){
            return res;
        }
        Deque<TreeNode> queue = new ArrayDeque<>();

        queue.offer(root);
        while(!queue.isEmpty()){
            int size = queue.size();
            List<Integer> temp = new ArrayList<>();
            while(size > 0){
                size--;
                TreeNode popped = queue.pop();
                temp.add(popped.val);
                if(popped.left != null){
                    queue.offer(popped.left);
                }
                if(popped.right != null){
                    queue.offer(popped.right);
                }
            }
            res.add(temp);
        }
        return res;
    }
}

File: Data Structures & Algorithms/candy/submission-4.py
class Solution:
    def candy(self, ratings: List[int]) -> int:
        
        n = len(ratings)
        
        left = [1]*n
        right = [1]*n

        left[0] = 1
        right[n-1] = 1

        for i in range(1, n):
            if ratings[i] > ratings[i-1]:
                left[i] = left[i-1] + 1
            # else:
            #     left[i] = 1

        for j in range(n-2, -1, -1):
            if ratings[j] > ratings[j+1]:
                right[j] = right[j+1] + 1
            # else:
            #     right[j] = 1 

        res = 0

        for i in range(n):
            res += max(left[i], right[i])

        return res                      



        